File: apps/Applications/ios/Local For Vocal/Local For Vocal/Sources/Views/Pages/InfluencerShopView.swift
import SwiftUI

#if canImport(UIKit)
    import UIKit
#endif

// MARK: - Theme Colors
extension Color {
    static let shopPrimary = Color(red: 189 / 255, green: 15 / 255, blue: 88 / 255)  // #bd0f58
    static let shopPrimaryDark = Color(red: 144 / 255, green: 11 / 255, blue: 66 / 255)  // #900b42
    static let shopBackground = Color(red: 253 / 255, green: 251 / 255, blue: 249 / 255)  // #fdfbf9
    static let shopSurface = Color.white
    static let shopTextMain = Color(red: 24 / 255, green: 17 / 255, blue: 20 / 255)  // #181114
    static let shopTextSecondary = Color(red: 137 / 255, green: 97 / 255, blue: 114 / 255)  // #896172
}

// MARK: - Product Detail for Display
struct ShopProductDetail: Identifiable {
    let id: String
    let name: String
    let imageUrl: String?
    let price: Double?
    let mrp: Double?
    let sellerName: String?

    init(from product: Product) {
        self.id = product.id
        self.name = product.name
        self.imageUrl = product.images.first  // images is [String], not optional
        self.price = product.price
        self.mrp = product.mrp
        self.sellerName = product.sellerName
    }

    init(from affiliateLink: AffiliateLink) {
        self.id = affiliateLink.productId
        self.name = affiliateLink.productName
        self.imageUrl = nil
        self.price = nil
        self.mrp = nil
        self.sellerName = nil
    }
}

struct InfluencerShopView: View {
    @Environment(\.presentationMode) var presentationMode
    @ObservedObject private var authManager = AuthManager.shared
    @State private var productDetails: [String: ShopProductDetail] = [:]
    @State private var isLoading = false
    @State private var stories: [Story] = []
    @State private var showStoryPlayer = false

    private let columns = [
        GridItem(.flexible(), spacing: 16),
        GridItem(.flexible(), spacing: 16),
    ]

    var body: some View {
        ZStack(alignment: .top) {
            Color.shopBackground
                .edgesIgnoringSafeArea(.all)

            ScrollView {
                VStack(spacing: 0) {
                    // Profile Section
                    profileSection

                    // Collections Carousel
                    if !uniqueCategories.isEmpty {
                        collectionsSection
                    }

                    // Product Feed Header
                    productHeader

                    // Product Grid
                    productGrid

                    Spacer(minLength: 100)
                }
            }

            // Sticky Header
            headerView
        }
        .navigationBarHidden(true)
        .toolbar(.hidden, for: .tabBar)  // Hide tab bar
        .task {
            await loadProductDetails()
            await loadStories()
        }
        .fullScreenCover(isPresented: $showStoryPlayer) {
            StoryPlayerWrapper(stories: stories, isPresented: $showStoryPlayer)
        }
    }

    // MARK: - Load Stories
    private func loadStories() async {
        do {
            let fetchedStories = try await APIService.shared.fetchMyStories()
            // Filter only active stories just in case
            await MainActor.run {
                self.stories = fetchedStories.filter { $0.isActive }
            }
        } catch {
            print("Failed to fetch stories: \(error)")  // Silent fail for UI
        }
    }

    // MARK: - Load Product Details
    private func loadProductDetails() async {
        guard !affiliateProducts.isEmpty else { return }
        isLoading = true

        for link in affiliateProducts {
            do {
                if let product = try await APIService.shared.fetchProductDetails(id: link.productId)
                {
                    await MainActor.run {
                        productDetails[link.productId] = ShopProductDetail(from: product)
                    }
                }
            } catch {
                // Use affiliate link data as fallback
                await MainActor.run {
                    if productDetails[link.productId] == nil {
                        productDetails[link.productId] = ShopProductDetail(from: link)
                    }
                }
            }
        }

        isLoading = false
    }

    // MARK: - Header
    private var headerView: some View {
        VStack(spacing: 0) {
            HStack {
                Button(action: {
                    presentationMode.wrappedValue.dismiss()
                }) {
                    Image(systemName: "arrow.left")
                        .font(.system(size: 20, weight: .medium))
                        .foregroundColor(.shopTextMain)
                        .frame(width: 40, height: 40)
                        .background(Color.white.opacity(0.9))
                        .clipShape(Circle())
                }

                Spacer()

                // Profile image + Shop name
                HStack(spacing: 8) {
                    // Profile image circle
                    if let imageUrl = authManager.currentUser?.profileImage,
                        !imageUrl.isEmpty,
                        let url = URL(string: imageUrl)
                    {
                        CachedAsyncImage(url: url) { image in
                            image
                                .resizable()
                                .aspectRatio(contentMode: .fill)
                        } placeholder: {
                            Image(systemName: "person.fill")
                                .font(.system(size: 12))
                                .foregroundColor(.shopTextSecondary)
                        }
                        .frame(width: 28, height: 28)
                        .clipShape(Circle())
                        .overlay(
                            Circle()
                                .stroke(Color.shopPrimary, lineWidth: 1.5)
                        )
                    } else {
                        Image(systemName: "person.fill")
                            .font(.system(size: 12))
                            .foregroundColor(.shopTextSecondary)
                            .frame(width: 28, height: 28)
                            .background(Color.gray.opacity(0.1))
                            .clipShape(Circle())
                    }

                    Text("\(authManager.currentUser?.name ?? "My")'s Shop")
                        .font(.system(size: 14, weight: .bold))
                        .foregroundColor(.shopTextMain)
                }

                Spacer()

                Button(action: {
                    shareInfluencerPage()
                }) {
                    Image(systemName: "square.and.arrow.up")
                        .font(.system(size: 20, weight: .medium))
                        .foregroundColor(.shopTextMain)
                        .frame(width: 40, height: 40)
                        .background(Color.white.opacity(0.9))
                        .clipShape(Circle())
                }
            }
            .padding(.horizontal, 16)
            .padding(.vertical, 8)
            .background(
                Color.shopBackground.opacity(0.95)
                    .background(.ultraThinMaterial)
            )

            Divider()
                .opacity(0.5)
        }
    }

    // MARK: - Profile Section
    private var profileSection: some View {
        VStack(spacing: 0) {
            Spacer().frame(height: 70)  // Space for header

            VStack(spacing: 16) {
                // Avatar with gradient border
                ZStack {
                    // Gradient Ring
                    if !stories.isEmpty {
                        Circle()
                            .fill(
                                LinearGradient(
                                    colors: [.shopPrimary, .orange],
                                    startPoint: .topTrailing,
                                    endPoint: .bottomLeading
                                )
                            )
                            .frame(width: 96, height: 96)
                    } else {
                        // Gray ring if no stories, or maybe just hidden?
                        // Keeping it for consistency but maybe lighter
                        Circle()
                            .stroke(Color.gray.opacity(0.2), lineWidth: 2)
                            .frame(width: 96, height: 96)
                    }

                    Circle()
                        .fill(Color.shopBackground)
                        .frame(width: 90, height: 90)

                    if let imageUrl = authManager.currentUser?.profileImage,
                        !imageUrl.isEmpty,
                        let url = URL(string: imageUrl)
                    {
                        CachedAsyncImage(url: url) { image in
                            image
                                .resizable()
                                .aspectRatio(contentMode: .fill)
                        } placeholder: {
                            Image(systemName: "person.fill")
                                .font(.system(size: 40))
                                .foregroundColor(.shopTextSecondary)
                        }
                        .frame(width: 84, height: 84)
                        .clipShape(Circle())
                    } else {
                        Image(systemName: "person.fill")
                            .font(.system(size: 40))
                            .foregroundColor(.shopTextSecondary)
                    }

                    // Verified badge
                    Image(systemName: "checkmark.circle.fill")
                        .font(.system(size: 20))
                        .foregroundColor(.shopPrimary)
                        .background(Circle().fill(Color.white).padding(2))
                        .offset(x: 32, y: 32)
                }
                .onTapGesture {
                    if !stories.isEmpty {
                        showStoryPlayer = true
                    }
                }

                // Name
                Text("\(authManager.currentUser?.name ?? "Influencer")'s Boutique")
                    .font(.system(size: 24, weight: .heavy))
                    .foregroundColor(.shopTextMain)

                // Bio
                Text("Sharing my favorite product recommendations ✨")
                    .font(.system(size: 14))
                    .foregroundColor(.shopTextSecondary)
                    .multilineTextAlignment(.center)
                    .padding(.horizontal, 40)

                // Product count badge
                HStack(spacing: 4) {
                    Image(systemName: "bag.fill")
                        .font(.system(size: 12))
                    Text("\(affiliateProducts.count) Products")
                        .font(.system(size: 13, weight: .semibold))
                }
                .foregroundColor(.shopPrimary)
                .padding(.horizontal, 16)
                .padding(.vertical, 8)
                .background(
                    Capsule()
                        .fill(Color.shopPrimary.opacity(0.1))
                )
            }
            .padding(.vertical, 24)
        }
    }

    // MARK: - Collections Section
    private var collectionsSection: some View {
        VStack(alignment: .leading, spacing: 12) {
            Text("COLLECTIONS")
                .font(.system(size: 11, weight: .bold))
                .foregroundColor(.shopTextSecondary)
                .tracking(0.5)
                .padding(.horizontal, 24)

            ScrollView(.horizontal, showsIndicators: false) {
                HStack(spacing: 12) {
                    ForEach(uniqueCategories, id: \.self) { category in
                        collectionItem(title: category)
                    }
                }
                .padding(.horizontal, 24)
            }
        }
        .padding(.bottom, 24)
    }

    private func collectionItem(title: String) -> some View {
        VStack(spacing: 8) {
            ZStack {
                RoundedRectangle(cornerRadius: 16)
                    .fill(
                        LinearGradient(
                            colors: [
                                Color.shopPrimary.opacity(0.8), Color.shopPrimary.opacity(0.4),
                            ],
                            startPoint: .topLeading,
                            endPoint: .bottomTrailing
                        )
                    )
                    .frame(width: 64, height: 64)

                Image(systemName: categoryIcon(for: title))
                    .font(.system(size: 24))
                    .foregroundColor(.white)
            }

            Text(title)
                .font(.system(size: 12, weight: .medium))
                .foregroundColor(.shopTextMain)
                .lineLimit(1)
        }
    }

    private func categoryIcon(for category: String) -> String {
        let lower = category.lowercased()
        if lower.contains("cloth") || lower.contains("fashion") || lower.contains("dress") {
            return "tshirt.fill"
        } else if lower.contains("beauty") || lower.contains("skin") || lower.contains("makeup") {
            return "sparkles"
        } else if lower.contains("shoe") || lower.contains("kick") {
            return "shoe.fill"
        } else if lower.contains("bag") || lower.contains("access") {
            return "bag.fill"
        } else if lower.contains("electr") || lower.contains("tech") {
            return "iphone"
        } else {
            return "tag.fill"
        }
    }

    // MARK: - Product Header
    private var productHeader: some View {
        HStack {
            Text("My Picks")
                .font(.system(size: 18, weight: .bold))
                .foregroundColor(.shopTextMain)

            Spacer()

            Text("\(affiliateProducts.count) items")
                .font(.system(size: 14))
                .foregroundColor(.shopTextSecondary)
        }
        .padding(.horizontal, 24)
        .padding(.bottom, 16)
    }

    // MARK: - Product Grid
    private var productGrid: some View {
        LazyVGrid(columns: columns, spacing: 16) {
            ForEach(affiliateProducts, id: \.productId) { link in
                NavigationLink(destination: ProductDetailView(productId: link.productId)) {
                    productCard(link: link)
                }
                .buttonStyle(PlainButtonStyle())
            }
        }
        .padding(.horizontal, 16)
    }

    private func productCard(link: AffiliateLink) -> some View {
        let detail = productDetails[link.productId]
        let imageUrl = detail?.imageUrl
        let price = detail?.price
        let sellerName = detail?.sellerName

        return VStack(alignment: .leading, spacing: 0) {
            // Product Image
            // Product Image
            Color.white
                .aspectRatio(1, contentMode: .fit)
                .overlay(
                    ZStack(alignment: .bottomLeading) {
                        if let urlString = imageUrl, let url = URL(string: urlString) {
                            CachedAsyncImage(url: url) { image in
                                image
                                    .resizable()
                                    .aspectRatio(contentMode: .fill)
                            } placeholder: {
                                Rectangle()
                                    .fill(Color.gray.opacity(0.1))
                                    .overlay(
                                        ProgressView()
                                            .progressViewStyle(
                                                CircularProgressViewStyle(tint: .shopPrimary))
                                    )
                            }
                            .frame(maxWidth: .infinity, maxHeight: .infinity)
                            .clipped()
                        } else {
                            Rectangle()
                                .fill(
                                    LinearGradient(
                                        colors: [Color.gray.opacity(0.1), Color.gray.opacity(0.2)],
                                        startPoint: .top,
                                        endPoint: .bottom
                                    )
                                )
                                .overlay(
                                    Image(systemName: "photo")
                                        .font(.system(size: 30))
                                        .foregroundColor(.gray.opacity(0.4))
                                )
                        }

                        // Curated badge
                        let curatorName =
                            (authManager.currentUser?.name ?? "Me").split(separator: " ").first.map(
                                String.init) ?? "Me"
                        Text("Curated by \(curatorName)")
                            .font(.system(size: 9, weight: .bold))
                            .foregroundColor(.white)
                            .textCase(.uppercase)
                            .tracking(0.3)
                            .padding(.horizontal, 6)
                            .padding(.vertical, 3)
                            .background(Color.black.opacity(0.6))
                            .cornerRadius(4)
                            .padding(8)
                    }
                )
                .clipped()

            // Details
            VStack(alignment: .leading, spacing: 4) {
                // Seller Name (Fixed Height)
                Text(sellerName ?? " ")
                    .font(.system(size: 11))
                    .foregroundColor(.shopTextSecondary)
                    .lineLimit(1)
                    .frame(height: 14, alignment: .leading)

                // Product Name (Fixed Height for 2 lines)
                Text(link.productName)
                    .font(.system(size: 13, weight: .bold))
                    .foregroundColor(.shopTextMain)
                    .lineLimit(2)
                    .multilineTextAlignment(.leading)
                    .frame(height: 32, alignment: .topLeading)  // Approx 2 lines + spacing line height

                HStack {
                    if let price = price {
                        Text("₹\(Int(price))")
                            .font(.system(size: 14, weight: .bold))
                            .foregroundColor(.shopPrimary)
                    } else {
                        Text("View Details")
                            .font(.system(size: 12, weight: .semibold))
                            .foregroundColor(.shopPrimary)
                    }

                    Spacer()

                    Image(systemName: "arrow.right.circle.fill")
                        .font(.system(size: 18))
                        .foregroundColor(.shopPrimary)
                }
                .frame(height: 20)  // Fixed height for price row
            }
            .padding(10)
            .frame(maxWidth: .infinity, alignment: .leading)
        }
        .background(Color.white)
        .cornerRadius(16)
        .shadow(color: Color.black.opacity(0.05), radius: 8, x: 0, y: 4)
    }

    // MARK: - Helpers
    private var affiliateProducts: [AffiliateLink] {
        authManager.currentUser?.affiliateLinks ?? []
    }

    private var uniqueCategories: [String] {
        // Extract unique categories from product names (simple heuristic)
        var categories: Set<String> = []
        for product in affiliateProducts {
            let name = product.productName.lowercased()
            if name.contains("dress") || name.contains("shirt") || name.contains("pant")
                || name.contains("lehenga") || name.contains("kurta") || name.contains("saree")
            {
                categories.insert("Fashion")
            } else if name.contains("bag") {
                categories.insert("Bags")
            } else if name.contains("shoe") || name.contains("sneaker") {
                categories.insert("Footwear")
            } else if name.contains("watch") || name.contains("jewelry") {
                categories.insert("Accessories")
            } else {
                categories.insert("All")
            }
        }
        return Array(categories).sorted()
    }

    private func shareInfluencerPage() {
        guard let user = authManager.currentUser,
            let referralCode = user.referralCode
        else { return }

        let shareText =
            "Check out my curated shop on Local For Vocal! 🛍️\nhttps://localforvocalstartup.com/shop/\(referralCode)"

        #if canImport(UIKit)
            guard let windowScene = UIApplication.shared.connectedScenes.first as? UIWindowScene,
                let rootViewController = windowScene.windows.first?.rootViewController
            else { return }

            var topController = rootViewController
            while let presented = topController.presentedViewController {
                topController = presented
            }

            let activityVC = UIActivityViewController(
                activityItems: [shareText], applicationActivities: nil)

            if let popover = activityVC.popoverPresentationController {
                popover.sourceView = topController.view
                popover.sourceRect = CGRect(
                    x: topController.view.bounds.midX, y: topController.view.bounds.midY, width: 0,
                    height: 0)
                popover.permittedArrowDirections = []
            }

            topController.present(activityVC, animated: true)
        #endif
    }
}

struct StoryPlayerWrapper: View {
    let stories: [Story]
    @Binding var isPresented: Bool
    @State private var currentIndex = 0

    var body: some View {
        ZStack {
            Color.black.edgesIgnoringSafeArea(.all)

            if !stories.isEmpty && currentIndex < stories.count {
                StoryView(story: stories[currentIndex]) {
                    // On Complete
                    if currentIndex < stories.count - 1 {
                        withAnimation {
                            currentIndex += 1
                        }
                    } else {
                        isPresented = false
                    }
                }
                .id(stories[currentIndex].id)  // Force recreate for fresh state
                .transition(.opacity)
            } else {
                // Fallback or end
                Color.black.onAppear {
                    isPresented = false
                }
            }
        }
    }
}

#Preview {
    InfluencerShopView()
}

File: apps/Applications/android/app/src/main/java/com/localforvocalstartup/app/ui/pages/InfluencerShopView.kt
package com.localforvocalstartup.app.ui.pages

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.GridItemSpan
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.lazy.grid.items
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import coil.compose.AsyncImage
import com.localforvocalstartup.app.data.manager.AuthManager
import com.localforvocalstartup.app.data.manager.NavigationManager
import com.localforvocalstartup.app.data.model.AffiliateLink
import com.localforvocalstartup.app.data.model.Product
import com.localforvocalstartup.app.data.model.Story
import com.localforvocalstartup.app.data.remote.NetworkClient

val ShopPrimary = Color(0xFFBD0F58)
val ShopPrimaryDark = Color(0xFF900B42)
val ShopBackground = Color(0xFFFDFBF9)
val ShopSurface = Color.White
val ShopTextMain = Color(0xFF181114)
val ShopTextSecondary = Color(0xFF896172)

data class ShopProductDetail(
    val id: String,
    val name: String,
    val imageUrl: String?,
    val price: Double?,
    val mrp: Double?,
    val sellerName: String?
) {
    constructor(product: Product) : this(
        id = product.id,
        name = product.name ?: product.title ?: "Unknown",
        imageUrl = product.images.firstOrNull(),
        price = product.price,
        mrp = product.mrp,
        sellerName = product.sellerName
    )
    constructor(link: AffiliateLink) : this(
        id = link.productId,
        name = link.productName,
        imageUrl = null,
        price = null,
        mrp = null,
        sellerName = null
    )
}

@Composable
fun InfluencerShopView() {
    val user = AuthManager.userState.collectAsState().value
    val affiliateProducts = user?.affiliateLinks ?: emptyList()
    
    var productDetails by remember { mutableStateOf<Map<String, ShopProductDetail>>(emptyMap()) }
    var isLoading by remember { mutableStateOf(false) }
    var stories by remember { mutableStateOf<List<Story>>(emptyList()) }
    var showStoryPlayer by remember { mutableStateOf(false) }

    LaunchedEffect(Unit) {
        if (affiliateProducts.isNotEmpty()) {
            isLoading = true
            val updatedDetails = mutableMapOf<String, ShopProductDetail>()
            for (link in affiliateProducts) {
                try {
                    val product = NetworkClient.apiService.getProductById(link.productId)
                    updatedDetails[link.productId] = ShopProductDetail(product)
                } catch (e: Exception) {
                    updatedDetails[link.productId] = ShopProductDetail(link)
                }
            }
            productDetails = updatedDetails
            isLoading = false
        }
        
        try {
            val fetchedStories = NetworkClient.apiService.fetchMyStories()
            stories = fetchedStories.filter { it.isActive }
        } catch (e: Exception) {
            // Ignore for visual consistency mock if network fails
        }
    }

    val uniqueCategories = remember(affiliateProducts) {
        val categories = mutableSetOf<String>()
        for (prod in affiliateProducts) {
            val name = prod.productName.lowercase()
            if (name.contains("dress") || name.contains("shirt") || name.contains("pant") || 
                name.contains("lehenga") || name.contains("kurta") || name.contains("saree")) {
                categories.add("Fashion")
            } else if (name.contains("bag")) {
                categories.add("Bags")
            } else if (name.contains("shoe") || name.contains("sneaker")) {
                categories.add("Footwear")
            } else if (name.contains("watch") || name.contains("jewelry")) {
                categories.add("Accessories")
            } else {
                categories.add("All")
            }
        }
        categories.toList().sorted()
    }

    Box(modifier = Modifier.fillMaxSize().background(ShopBackground)) {
        LazyVerticalGrid(
            columns = GridCells.Fixed(2),
            contentPadding = PaddingValues(top = 80.dp, bottom = 100.dp),
            horizontalArrangement = Arrangement.spacedBy(16.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            // Profile Section
            item(span = { GridItemSpan(2) }) {
                Column(
                    modifier = Modifier.fillMaxWidth().padding(vertical = 24.dp),
                    horizontalAlignment = Alignment.CenterHorizontally
                ) {
                    Box(
                        modifier = Modifier.clickable { 
                            if (stories.isNotEmpty()) showStoryPlayer = true 
                        },
                        contentAlignment = Alignment.Center
                    ) {
                        if (stories.isNotEmpty()) {
                            Box(
                                modifier = Modifier.size(96.dp).clip(CircleShape).background(
                                    Brush.linearGradient(listOf(ShopPrimary, Color(0xFFFFA500)))
                                )
                            )
                        } else {
                            Box(modifier = Modifier.size(96.dp).clip(CircleShape).background(Color.Gray.copy(alpha = 0.2f)))
                        }
                        
                        Box(modifier = Modifier.size(90.dp).clip(CircleShape).background(ShopBackground), contentAlignment = Alignment.Center) {
                            if (!user?.profileImage.isNullOrEmpty()) {
                                AsyncImage(
                                    model = user?.profileImage,
                                    contentDescription = "Avatar",
                                    contentScale = ContentScale.Crop,
                                    modifier = Modifier.size(84.dp).clip(CircleShape)
                                )
                            } else {
                                Icon(Icons.Default.Person, contentDescription = null, tint = ShopTextSecondary, modifier = Modifier.size(40.dp))
                            }
                        }
                        
                        Box(modifier = Modifier.size(96.dp), contentAlignment = Alignment.BottomEnd) {
                            Icon(Icons.Default.CheckCircle, contentDescription = null, tint = ShopPrimary, modifier = Modifier.size(24.dp).background(Color.White, CircleShape).padding(2.dp))
                        }
                    }
                    
                    Spacer(modifier = Modifier.height(16.dp))
                    Text("${user?.name ?: "Influencer"}'s Boutique", fontWeight = FontWeight.ExtraBold, fontSize = 24.sp, color = ShopTextMain)
                    Text("Sharing my favorite product recommendations ✨", fontSize = 14.sp, color = ShopTextSecondary, modifier = Modifier.padding(horizontal = 40.dp), textAlign = TextAlign.Center)
                    
                    Spacer(modifier = Modifier.height(16.dp))
                    Row(
                        modifier = Modifier
                            .background(ShopPrimary.copy(alpha = 0.1f), RoundedCornerShape(50))
                            .padding(horizontal = 16.dp, vertical = 8.dp),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Icon(Icons.Default.ShoppingBag, contentDescription = null, tint = ShopPrimary, modifier = Modifier.size(12.dp))
                        Spacer(modifier = Modifier.width(4.dp))
                        Text("${affiliateProducts.size} Products", color = ShopPrimary, fontSize = 13.sp, fontWeight = FontWeight.SemiBold)
                    }
                }
            }

            // Collections
            if (uniqueCategories.isNotEmpty()) {
                item(span = { GridItemSpan(2) }) {
                    Column(modifier = Modifier.padding(bottom = 24.dp)) {
                        Text("COLLECTIONS", fontSize = 11.sp, fontWeight = FontWeight.Bold, color = ShopTextSecondary, modifier = Modifier.padding(horizontal = 24.dp), letterSpacing = 0.5.sp)
                        Spacer(modifier = Modifier.height(12.dp))
                        LazyRow(contentPadding = PaddingValues(horizontal = 24.dp), horizontalArrangement = Arrangement.spacedBy(12.dp)) {
                            items(uniqueCategories) { category ->
                                Column(horizontalAlignment = Alignment.CenterHorizontally, verticalArrangement = Arrangement.spacedBy(8.dp)) {
                                    Box(
                                        modifier = Modifier.size(64.dp).clip(RoundedCornerShape(16.dp)).background(
                                            Brush.linearGradient(listOf(ShopPrimary.copy(alpha = 0.8f), ShopPrimary.copy(alpha = 0.4f)))
                                        ),
                                        contentAlignment = Alignment.Center
                                    ) {
                                        val icon = when {
                                            category.contains("Fashion") -> Icons.Default.Checkroom
                                            category.contains("Beauty") -> Icons.Default.AutoAwesome
                                            category.contains("Footwear") -> Icons.Default.DirectionsWalk
                                            category.contains("Bags") -> Icons.Default.ShoppingBag
                                            category.contains("Tech") -> Icons.Default.Smartphone
                                            else -> Icons.Default.LocalOffer
                                        }
                                        Icon(icon, contentDescription = null, tint = Color.White, modifier = Modifier.size(24.dp))
                                    }
                                    Text(category, fontSize = 12.sp, fontWeight = FontWeight.Medium, color = ShopTextMain, maxLines = 1)
                                }
                            }
                        }
                    }
                }
            }

            item(span = { GridItemSpan(2) }) {
                Row(
                    modifier = Modifier.fillMaxWidth().padding(horizontal = 24.dp, vertical = 16.dp),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.Bottom
                ) {
                    Text("My Picks", fontSize = 18.sp, fontWeight = FontWeight.Bold, color = ShopTextMain)
                    Text("${affiliateProducts.size} items", fontSize = 14.sp, color = ShopTextSecondary)
                }
            }

            // Product Grid
            items(affiliateProducts) { link ->
                val detail = productDetails[link.productId]
                val curatorName = user?.name?.split(" ")?.firstOrNull() ?: "Me"
                
                Column(
                    modifier = Modifier
                        .padding(horizontal = 16.dp)
                        .shadow(4.dp, RoundedCornerShape(16.dp))
                        .clip(RoundedCornerShape(16.dp))
                        .background(Color.White)
                        .clickable { NavigationManager.navigate("product/${link.productId}") }
                ) {
                    Box(modifier = Modifier.fillMaxWidth().aspectRatio(1f).background(Color.White)) {
                        if (!detail?.imageUrl.isNullOrEmpty()) {
                            AsyncImage(
                                model = detail?.imageUrl,
                                contentDescription = null,
                                contentScale = ContentScale.Crop,
                                modifier = Modifier.fillMaxSize()
                            )
                        } else {
                            Box(modifier = Modifier.fillMaxSize().background(Brush.verticalGradient(listOf(Color.Gray.copy(alpha = 0.1f), Color.Gray.copy(alpha = 0.2f)))), contentAlignment = Alignment.Center) {
                                Icon(Icons.Default.Image, contentDescription = null, tint = Color.Gray.copy(alpha = 0.4f), modifier = Modifier.size(30.dp))
                            }
                        }
                        
                        Box(contentAlignment = Alignment.BottomStart, modifier = Modifier.fillMaxSize().padding(8.dp)) {
                            Text("CURATED BY ${curatorName.uppercase()}", fontSize = 9.sp, fontWeight = FontWeight.Bold, color = Color.White, modifier = Modifier.background(Color.Black.copy(alpha = 0.6f), RoundedCornerShape(4.dp)).padding(horizontal = 6.dp, vertical = 3.dp), letterSpacing = 0.3.sp)
                        }
                    }
                    
                    Column(modifier = Modifier.padding(10.dp)) {
                        Text(detail?.sellerName ?: " ", fontSize = 11.sp, color = ShopTextSecondary, maxLines = 1, modifier = Modifier.height(14.dp))
                        Text(link.productName, fontSize = 13.sp, fontWeight = FontWeight.Bold, color = ShopTextMain, maxLines = 2, overflow = TextOverflow.Ellipsis, modifier = Modifier.height(32.dp))
                        Row(modifier = Modifier.height(20.dp).fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.CenterVertically) {
                            if (detail?.price != null) {
                                Text("₹${detail.price.toInt()}", fontSize = 14.sp, fontWeight = FontWeight.Bold, color = ShopPrimary)
                            } else {
                                Text("View Details", fontSize = 12.sp, fontWeight = FontWeight.SemiBold, color = ShopPrimary)
                            }
                            Icon(Icons.Default.ArrowCircleRight, contentDescription = null, tint = ShopPrimary, modifier = Modifier.size(18.dp))
                        }
                    }
                }
            }
        }

        // Header view matching iOS
        Column(modifier = Modifier.fillMaxWidth().background(ShopBackground.copy(alpha = 0.95f))) {
            Row(
                modifier = Modifier.fillMaxWidth().padding(horizontal = 16.dp, vertical = 8.dp),
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.SpaceBetween
            ) {
                IconButton(onClick = { NavigationManager.navigate("home") }, modifier = Modifier.size(40.dp).background(Color.White.copy(alpha = 0.9f), CircleShape)) {
                    Icon(Icons.Default.ArrowBack, contentDescription = "Back", tint = ShopTextMain, modifier = Modifier.size(20.dp))
                }
                
                Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                    if (!user?.profileImage.isNullOrEmpty()) {
                        AsyncImage(model = user?.profileImage, contentDescription = null, contentScale = ContentScale.Crop, modifier = Modifier.size(28.dp).clip(CircleShape))
                    } else {
                        Box(modifier = Modifier.size(28.dp).clip(CircleShape).background(Color.Gray.copy(alpha = 0.1f)), contentAlignment = Alignment.Center) {
                            Icon(Icons.Default.Person, contentDescription = null, tint = ShopTextSecondary, modifier = Modifier.size(12.dp))
                        }
                    }
                    Text("${user?.name ?: "My"}'s Shop", fontSize = 14.sp, fontWeight = FontWeight.Bold, color = ShopTextMain)
                }

                IconButton(onClick = { /* Share */ }, modifier = Modifier.size(40.dp).background(Color.White.copy(alpha = 0.9f), CircleShape)) {
                    Icon(Icons.Default.Share, contentDescription = "Share", tint = ShopTextMain, modifier = Modifier.size(20.dp))
                }
            }
            HorizontalDivider(color = Color.LightGray.copy(alpha = 0.5f))
        }
    }
}
